internal/pkg/service/stream/storage/level/local/writer/format/csv/fastcsv/fastcsv.go

```diff
@@ -0,0 +1,47 @@
+// Package fastcsv provides parallel writer/formatter for CSV records.
+//   - There is WritersPool with fixed number of writers.
+//   - The io.Writer.Write method of the out writer is always called with the whole CSV row.
+//   - There is no built-in synchronization of the output, so the io.Writer.Write method must be protected externally.
+//   - The values are always quoted, so it is possible to generate row in single pass, without scanning the column value.
+//   - The resulting CSV is therefore a bit larger, but we rely on the compression that may follow in the writers chain.
+package fastcsv
```

Comment:
Why is row by row writing of CSV more perfomant than stdlib `csv` package?

Reply:
Std CSV writer is not parallel, that's the main difference.

Here individual rows are prepared in parallel + and row buffer is reused.

Reply:
Maybe I don't understand it correctly. I have looked into code of `encoding/csv` and the code is simillar except of 2 things:

1. We use `row` buffer, that we fill with row information, and insert `row` into `out`. Why is this more performant than writing into `out` itself? Ofc, we can reuse buffer but we need sequentially copy buffer into out. 
2. The only thing I found working more optimized way is that we do this in go routines where we have pool of workers, so instead of writing sequential row by row, we do multiple rows instead.